lib/tokenlist.cpp

```diff
@@ -1768,6 +1768,10 @@ void TokenList::validateAst() const
             tok = tok->link();
             continue;
         }
+        if (isLambdaCaptureList(tok) && Token::simpleMatch(tok->astOperand1()->link()->next(), ".")) { // skip trailing return type
```

Comment:
Why do we skip for lambdas and not all functions?

Reply:
Because it's not necessary (see the added test) and there is no handy function for it?

Reply:
If we can have the valid AST for TRT in functions then we should have valid AST for TRT in lambdas as well.

Reply:
There is no consistency here:
`M::N::O h() { return {}; };` -> no AST
`auto h() -> M::N::O { return {}; };` -> AST
`decltype(s.i) h() { return {}; };` -> AST, looks wrong
`auto h() -> decltype(s.i) { return {}; };` -> AST

Reply:
More fun, no TRT AST for these:
~~~
auto h = []() -> M::N::O { return {}; };
auto h = []() -> decltype(s.i) { return {}; };
auto h = [&v]() -> decltype(v.begin()) { return {}; };
~~~
internalASTError:
`std::vector<int> v;
auto h = [&v]() -> decltype(v.begin()) { return {}; };`

Reply:
Since skipping validation is just a bandaid-type fix, I have reverted the corresponding commit.
See https://trac.cppcheck.net/ticket/11549 for the return type issues.
I suspect we don't even use the return type AST in case it is created.